src/buildstream/_sourcecache.py

```diff
@@ -148,7 +148,7 @@ def pull(self, source):
                 source.info("Remote cas ({}) does not have blob {} cached".format(remote, e.blob))
                 continue
             except CASError as e:
-                raise SourceCacheError("Failed to pull source {}: {}".format(display_key, e)) from e
+                raise SourceCacheError("Failed to pull source {}: {}".format(display_key, e), temporary=True) from e
```

Comment:
This one looks correct, just adding a note here because it explains something about the previous comment related to the same function `SourceCache.pull()`.

In this case we handle `CASError` after calling `CASCache.fetch_directory()`.

In this case, `CASCache.fetch_directory` *does* raise a `CASError` (because `CasCacheError` _is a `CASError`_), however in the previous case, `AssetRemote.fetch_directory()` does *not* raise a `CASError` but raises an `AssetCacheError` instead.

This probably explains why the previous code block incorrectly expects a `CASError`.